.travis.yml

```diff
@@ -0,0 +1,35 @@
+services:
+- docker
+
+before-install:
+  - echo "$DOCKER_PASSWORD" | docker login --username $DOCKER_USERNAME --password-stdin #login to docker hub
+  - curl https://cli-assets.heroku.com/install.sh | sh  #install heroku
+  - docker login --username=$HEROKU_LOGIN --password=$HEROKU_API_KEY registry.heroku.com  #login to registry.heroku.com
+  
+  
+before-script:
+  - docker build --target test --tag test .
+  - docker build --target production --tag $DOCKER_USERNAME/todo-app:latest .
+  - docker tag $DOCKER_USERNAME/todo-app:latest registry.heroku.com/$HEROKU_APP/web
```

Comment:
Did you set all of your environment variables in the Travis UI? It's a bit better to include them in this file so they are version controlled. The module 7 exercise explained you can do so for secret values too.